Composer/packages/client/src/components/WebChat/TranscriptFocusListener.tsx

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { FC, useCallback } from 'react';
+import { hooks } from 'botframework-webchat';
+import { Activity } from 'botframework-schema';
+import { useRecoilValue, useSetRecoilState } from 'recoil';
+
+import {
+  debugPanelActiveTabState,
+  debugPanelExpansionState,
+  dispatcherState,
+  rootBotProjectIdSelector,
+  webChatTrafficState,
+} from '../../recoilModel';
+import { WebChatInspectorTabKey } from '../../pages/design/DebugPanel/TabExtensions/types';
+
+const { useObserveTranscriptFocus } = hooks;
+
+export const TranscriptFocusListener: FC<{}> = () => {
```

Comment:
can this be a hook instead of a faceless component?

Reply:
Good suggestion, let me give that a shot.

Reply:
I don't think I can make them hooks, because they have to be within the Web Chat `<Composer />` component to have access to the APIs.

So I could make a component called `<WebChatHooksContainer />` or something and have that be a faceless component with all the hooks inside of it.

Reply:
Could the hooks just be invoked in the `WebChatComposer` component?

Reply:
Ahhh, that might work. Let me try that.

Reply:
yeah, that's what I meant, it can be hooks used in the exact same place they are rn.

Reply:
[WebChatComposer is created as a `React.memo`](https://github.com/microsoft/BotFramework-Composer/blob/main/Composer/packages/client/src/components/WebChat/WebChatComposer.tsx#L84) which hooks doesn't seem to like when I call them inside of the component. I think it was written as a memo to make it re-render only when certain props change.

Is there a way to make hooks work with a memo, or would I have to rewrite the component? In the latter case, I don't think it's worth rewriting a heavily used component for a small refactor.

Reply:
![image](https://user-images.githubusercontent.com/3452012/121087277-14ee0900-c799-11eb-8415-e36e97aec9fb.png)

This is what I get when I try to use them in `<WebChatComposer />`

Reply:
That's an error out of WebChat, probably due to using useObserveTranscriptFocus not within a Context.Provider provided by <Composer> from WebChat 

If you still want to keep them as hooks, you probably need to wrap them within a faceless component and put it as a child of <Composer>, having them as hooks might be better for unit tests, but at this point I'm fine with either way: separate faceless components, or two hooks wrapped in a faceless component

Reply:
Yup, went ahead with a faceless component wrapping both of the hooks